code/modules/modular_computers/file_system/programs/command/account_database.dm

```diff
@@ -91,46 +102,40 @@
 				"amount" = T.amount, \
 				"source_terminal" = T.source_terminal)))
 
-		if (trx.len > 0)
+		if(length(trx))
 			data["transactions"] = trx
 
-	var/list/accounts[0]
+	var/list/accounts = list()
 	for(var/M in SSeconomy.all_money_accounts)
 		var/datum/money_account/D = SSeconomy.get_account(M)
 		accounts.Add(list(list(\
 			"account_number"=D.account_number,\
 			"owner_name"=D.owner_name,\
 			"suspended"=D.suspended ? "SUSPENDED" : "")))
```

Comment:
Just send out all data with transactions and don't bother with `detailed_account_view` and switch between witch account is detailed in UI